src/cli/self_update/unix.rs

```diff
@@ -164,3 +164,17 @@ fn remove_legacy_paths() -> Result<()> {
     }
     Ok(())
 }
+
+fn remove_legacy_paths() -> Result<()> {
+    // Before the work to support more kinds of shells, we always inserted this
+    // which is now considered legacy
+    remove_legacy_path(format!(
+        "export PATH=\"{}/bin:$PATH\"\n",
+        shell::cargo_home_str()?
+    ))?;
+    // We had a release which used `source` rather than `.` which while widely
+    // supported, isn't actually POSIX, so we clean that up.
```

Comment:
I would prepend the comment with the Github issue number (it helps when the code is moved several times and git blame doesn't quickly provide something useful). It may also help to provide the rustup version, so that someone can decide in a year (or N years) that the remove_legacy_path is not needed anymore and delete the code.

Reply:
Mentioned both 1.23 and #2623